spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -1582,19 +1582,29 @@ def paste(self):
         text has 'CRLF' EOL chars
         """
         clipboard = QApplication.clipboard()
-        # This is here to prevent pasting mime data in the Editor.
-        # See issue 8566 for the details.
-        if clipboard.mimeData().hasUrls():
-            return
-        else:
-            text = to_text_string(clipboard.text())
-            if len(text.splitlines()) > 1:
-                eol_chars = self.get_line_separator()
-                text = eol_chars.join((text + eol_chars).splitlines())
-                clipboard.setText(text)
-            # Standard paste
-            TextEditBaseWidget.paste(self)
-            self.document_did_change(text)
+        text = to_text_string(clipboard.text())
+        # This is here to make copied files/folders to be pasted as text path.
+        # See issue 8566 and PR: 8576 for the details.
+        urls = []
+        if (not clipboard.mimeData().hasFormat('text/plain') and
+                clipboard.mimeData().hasUrls()):
+            urls = clipboard.mimeData().urls()
+            if len(urls) > 1:
+                text = ''.join('"' + url.toLocalFile().replace(osp.os.sep, '/')
+                               + '",' + '\n' for url in urls)

```

Comment:
This `\n` should be replaced `eol_chars` from `eol_chars = self.get_line_separator()` so they match the user's preferences.  

`eol_chars = self.get_line_separator()` should be define only once in the function.

Reply:
@Khalilsqu You should do this [here](https://github.com/spyder-ide/spyder/pull/8494/files#diff-e9841c32fa6af428b302e4ba35544d0eR890) in #8494 as well, btw.

Reply:
I need more investigation about this, should it go outside if statement or not! I will do it sometime tomorrow.